pkg/schema/annotations.go

```diff
@@ -146,6 +192,11 @@ func (n *DefaultAnnotation) NewTypeFromAnn() yamlmeta.Type {
 	return nil
 }
 
+// NewTypeFromAnn returns type information given by annotation. DescriptionAnnotation has no type information.
+func (n *DescriptionAnnotation) NewTypeFromAnn() yamlmeta.Type {
+	return nil
+}
+
```

Comment:
Can we massage `Annotation` into something more broadly suitable?... 🤔 

Here's an attempt:

I think we're "discovering" three (3) types of annotations:
- `TypingAnnotation` — contributes to the schema-type of the node
- `ValueHavingAnnotation` — contributes to the value of the node
- `DocumentingAnnotation` — contributes to the description of the node/type

And we'd break up `collectionAnnotation()` into three separate (very similar) functions that gather each collection.

Reply:
Great suggestion, added a function for this